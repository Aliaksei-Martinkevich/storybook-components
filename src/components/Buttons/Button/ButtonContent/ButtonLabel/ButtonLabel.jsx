import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './ButtonLabel.scss';


const ButtonLabel = ({ text, className = '' }) => (
  <span className={classNames('button-label', className)}>
    {text}
  </span>
);

ButtonLabel.propTypes = {
  text: PropTypes.string.isRequired,
  className: PropTypes.string,
};

export default ButtonLabel;
