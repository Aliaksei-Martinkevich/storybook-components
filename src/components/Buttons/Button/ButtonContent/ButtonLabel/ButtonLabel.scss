.button-label {
    overflow: hidden;
    text-overflow: clip;
    text-overflow: ellipsis;
    white-space: nowrap;
    text-align: left;
}