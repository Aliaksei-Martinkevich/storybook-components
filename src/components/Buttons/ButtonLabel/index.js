import React from 'react';
import PropTypes from 'prop-types';
import './index.css';


const ButtonLabel = ({text}) => (
<span className={'button-label'}>
  {text}
</span>
);

ButtonLabel.propTypes = {
  text: PropTypes.string.isRequired,
}

export default ButtonLabel;
