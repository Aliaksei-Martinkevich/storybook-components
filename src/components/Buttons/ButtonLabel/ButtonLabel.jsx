import React from 'react';
import PropTypes from 'prop-types';
import './ButtonLabel.scss';


const ButtonLabel = ({ text }) => (
  <span className={'button-label'}>
    {text}
  </span>
);

ButtonLabel.propTypes = {
  text: PropTypes.string.isRequired,
};

export default ButtonLabel;
