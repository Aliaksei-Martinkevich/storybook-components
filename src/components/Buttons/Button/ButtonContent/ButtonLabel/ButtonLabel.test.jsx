import React from 'react';
import renderer from 'react-test-renderer';
import ButtonLabel from './ButtonLabel.jsx';

it('Should render without fails', () => {
  const customClassName = 'class-name-example';
  const text = 'Some text';
  const component = renderer.create(<ButtonLabel className={customClassName} text={text} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
