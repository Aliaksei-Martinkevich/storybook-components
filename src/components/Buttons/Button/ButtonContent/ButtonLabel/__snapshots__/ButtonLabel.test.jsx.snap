// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button label Should render without fails 1`] = `
<span
  className="button-label class-name-example"
>
  Some text
</span>
`;
